### DISCUSSION TITLE
Local build

### DISCUSSION THREAD

**Discussion by IamRezaMousavi** [NickvisionApps/Denaro#487] [Q&A] [answered]
How can I build this app without `Visual Studio`? 
(In `linux` and `windows`)
Or how can build with `VSCode`?
Or with `make` or `cmake`?

**Comment by nlogozzo** [comment, marked as answer]
See https://github.com/NickvisionApps/Denaro/blob/main/CONTRIBUTING.md#developing-on-windows

TLDR: 

Because of WindowsAppSDK the application can only be built using Visual Studio on Windows

For the Linux version, you can use any IDE as you just need the dotnet CLI and `just` to build the app

**Reply by nlogozzo** [reply]
@fsobolev could you update Contributing.md across apps to include info about using just on linux?

**Reply by fsobolev** [reply]
> For the Linux version, you can use any IDE as you just need the dotnet CLI and just to build the app

Actually no, there are more dependencies that are needed if you want to build the app manually. All deps are:
1. dotnet 7
2. [just](https://github.com/casey/just)
3. blueprint-compiler
4. libadwaita **1.2+** and GTK **4.10**. Also, not sure, but devel packages of these libs may be required for blueprint-compiler
5. python 3.x
6. yelp-tools, gettext and itstool for documentation

GNOME Builder is recommended IDE because it will find flatpak manifest and the app can be build and run with single click, all dependencies will be handled and installed automatically.
There's VSCode extension to build flatpak, but currently it can't build Denaro without modifying manifest (https://github.com/bilelmoussaoui/flatpak-vscode/issues/212)
You can also test the app in WSL using `just run` command, but OpenSUSE Tumbleweed is probably the only distro supported in WSL that can be used, because of libadwaita and GTK versions requirements.
For any additional questions, join our matrix chat.

> could you update Contributing.md across apps to include info about using just on linux?

I will write all details there.

**Reply by IamRezaMousavi** [reply]
I use `Arch linux` and want to build all in one executable file without `flatpak`
Is it possible?
If yes, how?

**Reply by nlogozzo** [reply]
Like @fsobolev , as long as you have all the dependencies installed you can just clone the repo, cd into NickvisionMoney.GNOME and type `just run` and it'll build and run the app.

Similarly, there are denaro AUR packages in which you can look at the PKGBUILD and see how it's done there.

**Comment by IamRezaMousavi** [comment]
Thank you all
